src/transformers/models/unispeech/modeling_unispeech.py

```diff
@@ -947,7 +947,10 @@ def _conv_out_length(input_length, kernel_size, stride):
         return input_lengths
 
     def _get_feature_vector_attention_mask(self, feature_vector_length: int, attention_mask: torch.LongTensor):
-        output_lengths = self._get_feat_extract_output_lengths(attention_mask.sum(-1)).to(torch.long)
+        # Effectively attention_mask.sum(-1), but not inplace to be able to run
+        # on inference mode.
+        mask = attention_mask.cumsum(dim=-1)[:, -1]
```

Comment:
(nit) - could we call it `non_padded_lengths` - the idea here is to extract the sub sampled length from the "real non-padded" input length
```suggestion
        non_padded_lengths = attention_mask.cumsum(dim=-1)[:, -1]
```

Reply:
True, do you know any other way to do that operation ? it's very surprising that `.sum` is inplace, and I am scared that using `cumsum` instead is super wasteful.

Tried to grep it in our code, but I couldn't find anything of that sort.

Reply:
Hmm - don't really know to be honest...`torch.sum(...)` doesn't work either? But I think using `.cumsum(...)` is totally fine as well

Reply:
No, `torch.sum(..)` doesn't work.